Documentation/contributor-guide/features.md

```diff
@@ -0,0 +1,36 @@
+# Features 
+
+This document provides guidelines for adding and removing etcd features. The etcd features fall into two stages, experimental and stable. 
+
+## Experimental
+
+Any new feature should be added as an experimental feature. An experimental feature is characterized as below:
+- Associated with an issue with a clear need for such a feature.
+- Any configuration flags related to the implementation of the feature are prefixed with `experimental` e.g. `--experimental-feature-name`.
+- Any variable names related to the implementation of the feature are prefixed with `Experimental` e.g. `ExperimentalFeatureName`.
+- Might be buggy. Enabling the feature may not work as expected. 
+- Support for such a feature may be dropped at any time without notice.
+- Disabled by default.
+
+### Graduation to Stable
+
+It is important that experimental features don't get stuck in that stage. An experimental feature should move to the stable stage by using the criteria provided below: 
+- The feature was enabled for at least one major release or one year, whatever comes first. This time frame should be enough to find any potential bugs in the feature.
```

Comment:
Probably one `minor` release makes more sense here. For example, assuming we add an experimental feature `--experimental-abc` in 3.5, and it's disabled by default in 3.5.  It may be enabled by default in 3.6. Eventually we can graduate it in 3.7 if no any major issue is opened against the feature. 

If it's `major` release, then it means we can't graduate the feature until `5.0.`

Reply:
@ahrtr thanks and yes agree!! That's how I had in mind too :) but mixed up. I will update it.